molgenis-data/src/main/java/org/molgenis/data/support/DefaultEntity.java

```diff
@@ -275,7 +276,15 @@ public Entity getEntity(String attributeName)
 		EntityMetaData ref = attribute.getRefEntity();
 		ids = ids.stream().map(attribute.getDataType()::convert).collect(Collectors.toList());
 
-		return dataService.findAll(ref.getName(), IN(ref.getIdAttribute().getName(), ids));
+		String entityIdName = ref.getIdAttribute().getName();
+		Map<Object, Entity> entities = new HashMap<Object, Entity>();
```

Comment:
Fails for entities that refer to large number of other entities due to in memory storage, method should stream results back to caller.